hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/RecordLevelIndexSupport.scala

```diff
@@ -122,47 +123,78 @@ object RecordLevelIndexSupport {
   val INDEX_NAME = "RECORD_LEVEL"
 
   /**
-   * If the input query is an EqualTo or IN query on simple record key columns, the function returns a tuple of
+   * If the input query is an EqualTo or IN query on record key columns, the function returns a tuple of
    * list of the query and list of record key literals present in the query otherwise returns an empty option.
    *
    * @param queryFilter The query that need to be filtered.
    * @return Tuple of filtered query and list of record key literals that need to be matched
    */
-  def filterQueryWithRecordKey(queryFilter: Expression, recordKeyOpt: Option[String]): Option[(Expression, List[String])] = {
+  def filterQueryWithRecordKey(queryFilter: Expression, recordKeyOpt: Array[String]): Option[(Expression, List[String])] = {
+    val isComplexRecordKey = recordKeyOpt.length > 1
     queryFilter match {
+      // Handle EqualTo expressions
       case equalToQuery: EqualTo =>
         val attributeLiteralTuple = getAttributeLiteralTuple(equalToQuery.left, equalToQuery.right).orNull
         if (attributeLiteralTuple != null) {
           val attribute = attributeLiteralTuple._1
           val literal = attributeLiteralTuple._2
-          if (attribute != null && attribute.name != null && attributeMatchesRecordKey(attribute.name, recordKeyOpt)) {
-            Option.apply(equalToQuery, List.apply(literal.value.toString))
+          if (attribute != null && attribute.name != null && recordKeyOpt.contains(attribute.name)) {
+            val recordKeyLiteral = getRecordKeyLiteral(attribute, literal, isComplexRecordKey)
```

Comment:
How can we ensure all the consitituent record key values got involved in the expression?

Reply:
good point.. we should check if the predicate contains all of record key fields. Technically, we can support subset of record key fields as well but that involves pulling all records in RLI and checking for subset match. Something to followup. For now, we can just see if all record key fields are present or not. If not, then don't use RLI. What do you think?

Reply:
> For now, we can just see if all record key fields are present or not. If not, then don't use RLI. What do you think?

I kind of think we should also restrain all the expression type to `EQUALS`. For e.g, we have record key `col_0,col_1` and expression `col_0 >0 && col_1 <100`, you can not utilitize RLI in sucn use case.

Reply:
yes. 
- lets do RLI look up only when all fields in complex key gen are part of the predicates. 
- If I am not wrong, we already only support "IN" and "EQUAL"s. So, not sure what Danny is bringing up.

Reply:
We don't support range queries with RLI, so it will fallback to no data skipping (exxcept for partition pruning which is done before we read RLI).

Reply:
We need to check all the record key fields are got involved.

Reply:
Addressed